Backend/CarRentalApp/CarRentalApp/Services/Identity/UserService.cs

```diff
@@ -0,0 +1,33 @@
+﻿using CarRentalApp.Models.DTOs.Requests;
+using CarRentalApp.Models.Entities;
+using CarRentalApp.Services.Authentication;
+using CarRentalApp.Services.Data;
+
+namespace CarRentalApp.Services.Identity
+{
+    public class UserService
+    {
+        private UserRepository _userRepository;
+        private PasswordService _passwordService;
+
+        public UserService(UserRepository userRepository, PasswordService passwordService)
+        {
+            _userRepository = userRepository;
+            _passwordService = passwordService;
+        }
+
+        public async Task<User?> GetExistingUserAsync(UserLoginDTO user2Login)
+        {
+            return await _userRepository.GetByUsernameAsync(user2Login.Username);
+        }
+
+        public bool Validate(User existingUser, UserLoginDTO user2Validate)
```

Comment:
Why password matching with some external data is considered as a user validation?

Reply:
I would recommend to separate password validation from some validation (e.g. registration) and at least rename this method accordingly (e.g. ValidatePassword).